lib/resources/styles.css

```diff
@@ -79,6 +79,7 @@ main {
   overflow-y: scroll;
   padding: 20px 0 15px 30px;
   margin: 5px 20px 0 0;
+  visibility: hidden; /* shown by Javascript after scroll position restore */
```

Comment:
This all looks correct, but I can't help but wonder if there are places where this could fall off the rails. If you javascript script didn't run, or threw an exceptions for some reason before it could restore visibility?

Reply:
Yes, this is a risk.  If the javascript doesn't run due to an exception, the screen becomes blank (indeed, I experienced this when writing the code and accidentally introducing bugs).

The only thing I can think of to reduce that risk is to wrap all our JavaScript in the try of a try/finally with the visibility switch-on occurring in the finally block.

Reply:
I've wrapped the restore part that can be wrapped in a try-finally block, and moved it ahead of all the other javascript initialization.  This means the page visibly appears slightly faster and should be more resilient to exceptions.